enterprise/server/remote_execution/execution_server/execution_server_test.go

```diff
@@ -0,0 +1,157 @@
+package execution_server_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/go-redis/redis/v8"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"gorm.io/gorm"
+
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/remote_execution/execution_server"
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/scheduling/task_router"
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/testutil/testredis"
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/util/redisutil"
+	"github.com/buildbuddy-io/buildbuddy/server/interfaces"
+	"github.com/buildbuddy-io/buildbuddy/server/tables"
+	"github.com/buildbuddy-io/buildbuddy/server/testutil/testenv"
+
+	"github.com/buildbuddy-io/buildbuddy/proto/remote_execution"
+)
+
+type schedulerServerMock struct {
+	interfaces.SchedulerService
+}
+
+func (s *schedulerServerMock) CancelTask(ctx context.Context, taskID string) (bool, error) {
+	return true, nil
+}
+
+func executionServer(t *testing.T, env *testenv.TestEnv) *execution_server.ExecutionServer {
+	redisTarget := testredis.Start(t).Target
+	rdb := redis.NewClient(redisutil.TargetToOptions(redisTarget))
+	env.SetRemoteExecutionRedisClient(rdb)
+	env.SetRemoteExecutionRedisPubSubClient(rdb)
+
+	router, err := task_router.New(env)
+	if err != nil {
+		t.Fatal(err)
+	}
+	env.SetTaskRouter(router)
+
+	scheduler := &schedulerServerMock{}
+	env.SetSchedulerService(scheduler)
+
+	s, err := execution_server.NewExecutionServer(env)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return s
+}
+
+func createExecution(t *testing.T, db *gorm.DB, execution *tables.Execution) {
+	err := db.Create(execution).Error
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestCancel(t *testing.T) {
+	env := testenv.GetTestEnv(t)
+	ctx := context.Background()
+	s := executionServer(t, env)
+
+	// Create Execution rows to be canceled
+	db := env.GetDBHandle().DB(ctx)
+	testUUID, err := uuid.NewRandom()
+	if err != nil {
+		t.Fatal(err)
+	}
+	testInvocationID := testUUID.String()
+
+	executionID := "blobs/1111111111111111111111111111111111111111111111111111111111111111/100"
+	execution := &tables.Execution{
+		ExecutionID:  executionID,
+		InvocationID: testInvocationID,
+		Stage:        int64(remote_execution.ExecutionStage_EXECUTING),
+	}
+	createExecution(t, db, execution)
+
+	numCanceled, err := s.Cancel(ctx, testInvocationID)
+	assert.Nil(t, err)
```

Comment:
nit, it gives a clearer error message if you use `assert.NoError`

I also would recommend leaning towards `require` from the testify package instead of `assert`, since require() will fail the test and not move on to future assertions. In this case if there is an error then the assert on numCanceled is expected, and the failed assertion will add more noise to the test.

Reply:
Done